proxy/src/main.rs

```diff
@@ -108,6 +123,37 @@ async fn main() -> anyhow::Result<()> {
         )
         .get_matches();
 
+    if let Some(dir) = arg_matches.value_of("datadir") {
+        // change into the data directory.
+        println!("Working in {}", dir);
+        std::env::set_current_dir(PathBuf::from(dir))?;
+    }
+
+    if arg_matches.is_present("daemonize") {
+        println!("daemonizing...");
+
+        // There should'n be any logging to stdin/stdout. Redirect it to the main log so
+        // that we will see any accidental manual fprintf's or backtraces.
+        let log_file = zenith_utils::logging::init("proxy.log", true)?;
+        let stdout = log_file.try_clone().unwrap();
+        let stderr = log_file;
```

Comment:
I firmly believe that being able to redirect logs to a file is orthogonal to daemonization. Sure, daemons can't write logs to stdout, but it doesn't necessarily mean that it shouldn't be possible to choose this option whenever we'd like to (this rant mostly refers to pageserver/safekeeper/etc, but this where you took this from).